docs/content/src/learn/guides/integration-patterns/content-based-routing/README.md

```diff
@@ -0,0 +1,99 @@
+# Content-Based Routing
+
+## About
+
+Content-based routing is an integration pattern where the message received from a client is routed to different channels/paths/endpoints based on the content of the message. This guide demonstrates a simple content-based routing scenario where, based on the company name provided in the request, the relevant company’s service endpoint is called to get the stock quote.
+
+## What you'll build
+
+We shall have a service ‘stockQuote’ which accepts an http request from a client. The client appends as a query parameter in the request, the company name of which he wants to know the price of the stock. The stockQuote service identifies the company, routes the request to the relevant company’s stock quote service, obtain the response from the company’s service and returns the response to the client. If the company name is not available in the request, the service will simply respond with a 400 - Bad Request.
+
+![cbr](../../assets/img/cbr.png)
+
+## Prerequisites
+
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install the Ballerina IDE plugin for [VS Code](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina)
+	
+## Implementation
+
+* Create a new ballerina project named ‘content-based-routing’.
+
+```bash
+$ ballerina new content-based-routing
+```
+
+* Navigate to the directory content-based-routing.
+
+* Add a new module named ‘stockquote_service’ to the project.
+
+```bash
+$ ballerina add stockquote_service
+```
+
+* Open the project with VSCode. The project structure will be like the following.
+
+```shell
+.
+├── Ballerina.toml
+└── src
+    └── stockquote_service
+        ├── main.bal
+        ├── Module.md
+        ├── resources
+        └── tests
+            ├── main_test.bal
+            └── resources
+```
+
+We can remove the file `main_test.bal` at the moment, since we're not writing any tests for our service.
+
+First let's create the services that we shall be using as backend endpoints. 
+
+* Create a new file named `ibmService.bal` file under 'stockquote_service' with the following content.
+
+**ibmService.bal**
+<!-- INCLUDE_CODE: src/stockquote_service/ibmService.bal -->
+
+This is simply a service that will run on port 8081 responding a text payload `10000.00`.
```

Comment:
It's better if we can give a more detailed summary here.

Reply:
This service is only there to respond a text payload "10000.00". This is kind of a mock service.